src/main/scala/com/amazon/deequ/schema/ColumnName.scala

```diff
@@ -0,0 +1,100 @@
+/**
+  * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License"). You may not
+  * use this file except in compliance with the License. A copy of the License
+  * is located at
+  *
+  *     http://aws.amazon.com/apache2.0/
+  *
+  * or in the "license" file accompanying this file. This file is distributed on
+  * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+  * express or implied. See the License for the specific language governing
+  * permissions and limitations under the License.
+  *
+  */
+
+package com.amazon.deequ.schema
+
+object ColumnName {
+
+  type Sanitized = Either[SanitizeError, String]
+
+  /**
+    * Sanitizes the input column name by ensuring that it is escaped with backticks.
+    *
+    * The resulting String is the escaped input column name, which is safe to use in
+    * any Spark SQL statement.
+    */
+  def sanitizeForSql(columnName: String): Sanitized =
+    if (columnName == null) {
+      Left(NullColumn)
+
+    } else {
+      val rawColumnName = extractColumnName(columnName)
+      if (rawColumnName.contains("`")) {
+        Left(ColumnNameHasBackticks(columnName))
+      } else {
+        Right(s"`$rawColumnName`")
+      }
+    }
+
+  /** Extracts the original column name from a possibly sanitized column name. */
+  @inline
+  private[this] def extractColumnName(name: String): String =
+    name.slice(
+      if (hasStartingTick(name)) 1 else 0,
+      name.length - (if (hasEndingTick(name)) 1 else 0)
+    )
+
+  /** True iff the input string starts with a backtick. False otherwise. */
+  @inline
+  private[this] def hasStartingTick(s: String): Boolean =
+    s.startsWith("`")
+
+  /** True iff the input string ends with a backtick. False otherwise. */
+  @inline
+  private[this] def hasEndingTick(s: String): Boolean =
+    s.endsWith("`")
+
+  /** Obtains the `String` value if `Right` or throws the `SanitizeError` if `Left`. */
+  def getOrThrow(x: Sanitized): String = x match {
+    case Left(error) => throw error
+    case Right(str) => str
+  }
+
+  /**
+    * Obtains the `String` pair if both are `Right`, otherwise throws the error(s).
+    *
+    * If only one of the sanitizations failed, then a `SanitizeError` type is thrown.
+    * If both fail, then an `IllegalArgumentException` is thrown and its message contains
+    * both of the `SanitizeError` messages.
+    */
+  def getOrThrow(x: (Sanitized, Sanitized)): (String, String) = x match {
```

Comment:
where exactly is this used? I'm not so happy with a method that could throw an exception from some place in the code.

Reply:
This is used in `sanitize`, which is used all-over the place: whenever there's a column name that is going into a Spark SQL expression. I would prefer to not throw exceptions too. However, the code written as-is assumes that columns would always be sanitized. As a result, the various Spark SQL-calling code doesn't have return types that encode an error: they always assume that the computation will succeed.

As it currently stands, `deequ`'s current behavior is to already throw an exception when using an unsanitzied column name with special characters during Spark SQL execution: it would be a Spark SQL `SyntaxError`. So this PR doesn't stray too far from existing functionality. In fact, it would only throw an exception now if the column name could not be sanitized (i.e. has backticks  in the name -- not a pair in the surrounding/escaped position).